test/mocks.h

```diff
@@ -31,13 +32,48 @@ struct MockLogger : public Logger {
   void Trace(uint64_t, uint64_t, ot::string_view) const noexcept override {}
 };
 
+// `Journal` is an implementation of `Logger` that appends logged records to an
```

Comment:
Maybe make the MockLogger a better fit for testing purposes instead of introducing a new logger-thing?